basxconnect/mailer_integration/download_data.py

```diff
@@ -0,0 +1,69 @@
+from typing import NamedTuple
+
+from basxconnect.core import models
+from basxconnect.mailer_integration.abstract.abstract_datasource import Datasource
+from basxconnect.mailer_integration.abstract.abstract_mailer_person import MailerPerson
+from basxconnect.mailer_integration.models import Interest, MailingPreferences
+
+
+class SynchronizationResult(NamedTuple):
+    new_persons: int
+    total_synchronized_persons: int
+
+
+def download_persons(datasource: Datasource) -> SynchronizationResult:
+    mailer_persons = datasource.get_persons()
+    datasource_tag = _get_or_create_tag(datasource.tag())
+    new_persons = 0
+    for mailer_person in mailer_persons:
+        matching_persons = list(
+            models.Person.objects.filter(
+                primary_email_address__email=mailer_person.email(), deleted=False
+            ).all()
+        )
+        if len(matching_persons) == 0:
+            _save_person(datasource_tag, mailer_person)
+            new_persons += 1
+        else:
+            # if the downloaded email address already exists in our system, update the mailing preference of all persons
+            # that use this email address (as primary email address), without creating a new person in the database
+            for person in matching_persons:
+                _save_mailing_preferences(person, mailer_person)
+
+    return SynchronizationResult(new_persons, len(mailer_persons))
+
+
+def _is_new_person(
+    person: MailerPerson,
+) -> bool:
+    return not models.Person.objects.filter(
+        primary_email_address__email=person.email(), deleted=False
```

Comment:
Here is another "synchronization" question (conceptually, nothing wrong with the code): What happens if we have an email-address stored but it is not assigned as primary email? freebeat noted that he understands that the mailchimp email address is a unique identifier/primary key/identity. But in our system this is not the case. I think we should add a note to the sync code to make that a bit more clear, I guess it should also be part of our discussion in the next meeting.

As a side note: Thinking about "identity" is crucial when we deal with external data (even if it is just a CSV file). Determining if two things are "the same" or are "equal" is often not straight forward.